### PR TITLE
8296283: JUnit5 tests using Params API fails to compile

### PR DESCRIPTION
**Pull request by arapte**
This issue was uncovered when running the test added in PR: https://github.com/openjdk/jfx/pull/910
The JUnit5 tests using Params API fails to compile.

Fix:
Change dependency of `junit-jupiter-params` from runtime to compile time.
From: https://junit.org/junit5/docs/current/user-guide/#overview-what-is-junit-5: it can be understood that packages of JUnit Jupiter other than `TestEngine` are required for test compilation and
JUnit Platform serves as a foundation for [launching testing frameworks](https://junit.org/junit5/docs/current/user-guide/#launcher-api)

It seems that changing dependency of this one package is sufficient for this fix.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8296283](https://bugs.openjdk.org/browse/JDK-8296283): JUnit5 tests using Params API fails to compile

### Reviewers
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Andy Goryachev](https://openjdk.org/census#angorya) (@andy-goryachev-oracle - Author) 🔄 Re-review required (review applies to [64ba3a96](https://git.openjdk.org/jfx/pull/939/files/64ba3a9655359fe38dfb80ed8d51d047bd057ee0))

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx pull/939/head:pull/939` \
`$ git checkout pull/939`

Update a local copy of the PR: \
`$ git checkout pull/939` \
`$ git pull https://git.openjdk.org/jfx pull/939/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 939`

View PR using the GUI difftool: \
`$ git pr show -t 939`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/939.diff">https://git.openjdk.org/jfx/pull/939.diff</a>

</details>

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1988,10 +1988,10 @@ allprojects {
         testImplementation group: "org.hamcrest", name: "hamcrest-core", version: "1.3"
         testImplementation group: "org.junit.jupiter", name: "junit-jupiter", version: "5.8.1"
         testImplementation group: "org.junit.jupiter", name: "junit-jupiter-api", version: "5.8.1"
+        testImplementation group: "org.junit.jupiter", name: "junit-jupiter-params", version: "5.8.1"
         testImplementation group: "org.opentest4j", name: "opentest4j", version: "1.2.0"
         testRuntimeOnly group: "org.apiguardian", name: "apiguardian-api", version: "1.1.2"
         testRuntimeOnly group: "org.junit.jupiter", name: "junit-jupiter-engine", version: "5.8.1"
-        testRuntimeOnly group: "org.junit.jupiter", name: "junit-jupiter-params", version: "5.8.1"
         testRuntimeOnly group: "org.junit.platform", name: "junit-platform-commons", version: "1.8.1"
```
Comment (kevinrushforth) [inline build.gradle:1995]: Have you checked whether the `junit-platform-commons` jar file, which also has public JUnit5 API, needs the same treatment? I might recommend also moving that one to `testImplementation` as well.

**Review by kevinrushforth — APPROVED** [review]
Looks good, pending my question about `junit-platform-commons`.

File: build.gradle (hunk 1 above)
Comment (arapte) [reply build.gradle:1995]: Hi Kevin, Thanks for the pointer. 
As we checked that the jupiter-platform classes are used for test framework development, a change for them is not required for now.

**Comment by arapte** [comment]
@kevinrushforth 
Added a test as you suggested offline.
The checks inside test may not be required, please suggest if any modifications required.

**Comment by andy-goryachev-oracle** [comment]
We have other tests which use `@RunWith(Parameterized.class)` just fine (for example, SkinMemoryLeakTest).  Is this about mixing junit4 and junit5 classes?

Do we need to make changes in other tests?

**Comment by nlisker** [comment]
It could be a good idea in a follow-up PR to move the version numbers to a constant so that we don't need to update each separately, like the Junit "5.8.1" version.

**Comment by arapte** [comment]
> We have other tests which use `@RunWith(Parameterized.class)` just fine (for example, SkinMemoryLeakTest). Is this about mixing junit4 and junit5 classes?

It is not about mixing. The JUnit5 test added in PR https://github.com/openjdk/jfx/pull/910 which uses annotations from `org.junit.jupiter.params` failed to compile on my local machine throwing following errors:
```
error: package org.junit.jupiter.params does not exist
error: package org.junit.jupiter.params.provider does not exist
error: cannot find symbol @ValueSource
error: cannot find symbol @ParameterizedTest 
```
This PR fixes that compilation error.
I am not sure how it worked on GHA.

> Do we need to make changes in other tests?

It won't need any changes in existing tests.

File: modules/javafx.base/src/test/java/test/JUnit5Test.java [hunk 2]
```diff
@@ -39,4 +43,30 @@ void junit5ShouldWork() {
...
+        }
+        callCount++;
+
+        assertTrue(match, "Received incorrect value as parameter");
+        assertTrue(callCount <= intValues.length, "Test function called more than number of ValueSources");
+        if (callCount == intValues.length) {
+            for (int i : intValues) {
+                if (i != 0) {
+                    fail("Test method not called for Value " + i);
+                }
```
Comment (kevinrushforth) [inline modules/javafx.base/src/test/java/test/JUnit5Test.java:68]: Minor: could simplify this to:

```
    assertEquals(0, i, "Test method not called for Value " + i);
```

File: modules/javafx.base/src/test/java/test/JUnit5Test.java [hunk 3]
```diff
@@ -39,4 +43,30 @@ void junit5ShouldWork() {
         assertNotNull(this);
         System.err.println("JUnit 5 test working!");
     }
+
+    static int callCount;
+    static int[] intValues = {1, 2, 3};
+    @ValueSource(ints = {1, 2, 3})
```
Comment (kevinrushforth) [inline modules/javafx.base/src/test/java/test/JUnit5Test.java:49]: Minor: maybe add a blank line before this line?

**Review by kevinrushforth — APPROVED** [review]
The test looks good. I left a couple minor comments.

**Review by andy-goryachev-oracle — APPROVED** [review]
thank you for clarifications.

**Comment by kevinrushforth** [comment]
> We have other tests which use `@RunWith(Parameterized.class)` just fine (for example, SkinMemoryLeakTest). Is this about mixing junit4 and junit5 classes?
> 
> Do we need to make changes in other tests?

No, this bug is about specific APIs that are part of the `junit-jupiter-params` jar file. Specifically, the `ValueSource` and `ParameterizedTest` annotations not being available at compile time in all cases. It works accidentally without the `buld.gradle` fix for open builds (e.g., GHA) and fails in our CI build without this change.

**Comment by kevinrushforth** [comment]
> It could be a good idea in a follow-up PR to move the version numbers to a constant so that we don't need to update each separately, like the Junit "5.8.1" version.

Yes, this is a good idea. @arapte can you file a follow-on issue?

File: modules/javafx.base/src/test/java/test/JUnit5Test.java (hunk 2 above)
Comment (arapte) [reply modules/javafx.base/src/test/java/test/JUnit5Test.java]: Changed the assert.

File: modules/javafx.base/src/test/java/test/JUnit5Test.java (hunk 3 above)
Comment (arapte) [reply modules/javafx.base/src/test/java/test/JUnit5Test.java:50]: Added new line.

**Pull request closed without merging**